File: javascript/Hash/flattenHash/flattenHash.js
// Given a hash, return a flat array containing all the hash’s keys and values.

// Input: {“a” => 1, “b” => 2, “c” => 3, “d” => 4}
// Output: [“a”, 1, “b”, 2, “c”, 3, “d”, 4]

const flattenHash = function (hash) {
  let result = []
  for (let number in hash) {
    result.push(number)
    result.push(hash[number])
  }
  return result
}

console.log(flattenHash({a: 1, b: 2, c: 3, d: 4}))

File: ruby/algo_ladder/flatten_hash/flatten_hash.rb
# Given a hash, return a flat array containing all the hash’s keys and values.

# Input: {“a” => 1, “b” => 2, “c” => 3, “d” => 4}
# Output: [“a”, 1, “b”, 2, “c”, 3, “d”, 4]

def flatten_hash (input_hash)
  result = []

  input_hash.each do |key, value|
    result << key
    result << value
  end
  return result
end

p flatten_hash({"a" => 1, "b" => 2, "c" => 3, "d" => 4})